R/ab_scenario.R

```diff
@@ -1,11 +1,13 @@
 #' Generate A/B Street Scenario files/objects
 #'
-#' @param houses Polygons where trips will originate (`sf` object)
+#' @param od Origin destination data
+#' @param zones Zones with IDs that match the desire lines and class `sf`
+#' @param subpoints Polygons where trips will originate (`sf` object)
 #' @param buildings Buildings where trips will end represented as `sf` object
 #' @param desire_lines Origin-Destination data represented as `sf`
```

Comment:
`desire_lines` is one of the only params that I have any idea how it actually looks. How does `pop_var` match up with `desire_lines`? Are they both lists guaranteed to have the same number of members, and they just match up by index?

For `od Origin destination data`, is this a list of tuples like `("zone1", "zone2", 500)` or a 2D OD matrix of some sort or something else? How are you supposed to infer the shape of parameters in R?

Reply:
OD is a data frame, with an optional geometry column if it has been converted into a geographic line, representing movement between origin zones represented in the 1st column and destinations in the 2nd. There is no formal class definition of this dataset but it is well documented in the `od` package and an ongoing (although stagnant) discussion about creating a formal class definition that could provide the clarity that's needed: https://github.com/ITSLeeds/od/issues/20

For now it just relies on the user having read the documentation and knowing that it will report and error if incorrect data is passed.